modules.d/00systemd/module-setup.sh

```diff
@@ -228,13 +228,23 @@ install() {
 
     mkdir -p "$initdir/etc/systemd"
     # We must use a volatile journal, and we don't want rate-limiting
+    # and want early startup messages to go everywhere kernel buffer does
     {
         echo "[Journal]"
         echo "Storage=volatile"
         echo "RateLimitInterval=0"
         echo "RateLimitBurst=0"
+		echo "ForwardToConsole=yes"
+		echo "ForwardToKMsg=yes"
+        echo "MaxLevelKMsg=info"
     } >> "$initdir/etc/systemd/journald.conf"
 
+    mkdir -p "$initdir/etc/sysctl.d"
+    # Also disable rate-limiting at kernel ring buffer level
+    {
+        echo "kernel.printk_devkmsg = on"
+    } >> "$initdir/etc/sysctl.d/10-dont-ratelimit-kmsg.conf"
```

Comment:
the biggest problem here is that the default for this setting (`ratelimit`) will not get restored after this has run. I don't know how to resolve that. Really I only want to change it to `on` during initramfs and after that drop it back to whatever it was set to before (either the default  or the value from the kernel command line).